### ISSUE TITLE
Add support for Finalizer interface

### ISSUE THREAD

**Issue by j-fischer** [j-fischer/rflib#11]
See https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_transaction_finalizers.htm for more details.

**Comment by mikbranchaud** [comment]
Is it strictly necessary to use the Finalizer pattern to perform any logging from a queueable class?  Or is this just in case you don't/can't use try/catch/finally and exit gracefully on an exception?  It seems like if the PE is published immediately, the Finalizer is not needed?

I have an apex class with a lot of utility functions in it, Utility.  I have another class which implements queueable but where all the logic is also in the Utility class.  So a function in the Utility class calls the queueable class which in turn calls a different function in the Utility class.  The LOGGER is declared in the Utility class I am using the batched logging pattern with try/catch/finally.  I see the log PE being created in the initial Utility class function but nothing from after the queueable class invokes the Utility class function.  We have a version of rflib from before the Finalizer implementation so I am not sure if I need to upgrade to support logging after a queueable  call.

**Comment by j-fischer** [comment]
Great question and thanks for using the library. 

Everything you said was spot on. The Finalizer interface is for uncaught exceptions that would interrupt the transaction. If you use the rflib_Logger in your Queueable or utility classes, it will just continue to work like it did before. You can just choose if you want to add the finalizer as an addition for any unexpected errors. One thing to keep in mind is that there are catchable and uncatchable exceptions in Apex and the Finalizer interface will be able to capture and log both types of exceptions. 

To sum it up: No need to upgrade, but if you do, it would not hurt to add it just in case there are any unexpected outcomes of a Queueable run.  

Hope that answered your question :)
